src/labone/nodetree/node.py

```diff
@@ -45,6 +45,50 @@
 T = t.TypeVar("T")
 
 
+class AnnotatedValueOrResultNodePromise:
```

Comment:
issue: The custom class should come from the core and should be named something like `AnnotatedValuePromise` and be used everywhere we return an annotated value asynchronously.

Doing it only for the nodetree would be an ugly hotfix.